miraclelinux/meta-debian
# ncurses: Update version from 6.1+20181013-2+deb10u3 to 6.1+20181013-2+deb10u4

# Purpose of pull request

This version fixes following CVEs.
- CVE-2020-19189

DLA: DLA 3586-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build and run the qemuarm64 image
2. Run commands to check the ncurses working

### Build and run the qemuarm64 image
Add the following to local.conf.
```
IMAGE_INSTALL_append = " ncurses"
```
```
$ bitbake core-image-minimal
$ runqemu qemuarm64 nographic
```

### Run commands to check the ncurses working
```
# tset -V
# tput -V
# tset -q
# ls -F /etc/terminfo/*
# export TERM=xterm
# tset -q
# ls --color -F /
# tput setaf 1
# echo "Hello World !!"
# tput sgr0
```

## Test result
### Build package
Build succeeded.
```
$ bitbake ncurses
Parsing recipes: 100% |################################################################################################################| Time: 0:00:29
Parsing of 1362 .bb files complete (0 cached, 1362 parsed). 2382 targets, 81 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.8"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-ncurses_6.1+20181013-2+deb10u4:675ac4c5e94bca9e71bc65504641cdf6d891f1ad"
meta-debian-extended = "HEAD:9e95cec8547eb380bce0c68df227b67043e233fe"
meta-emlinux         = "HEAD:4dd2259ccc4eaaa6dece7f164eb2827bd09eadc4"
meta-emlinux-private = "HEAD:8207525289999d2c41983eb8155127b62621cdb1"

Initialising tasks: 100% |#############################################################################################################| Time: 0:00:01
Sstate summary: Wanted 306 Found 0 Missed 306 Current 120 (0% match, 28% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 1969 tasks of which 1483 didn't need to be rerun and all succeeded.
```

### Package test
The tset and tput commands are working.
![image](https://github.com/miraclelinux/meta-debian/assets/118246107/94f73d04-1905-40c0-8ea9-13bbb941fe35)
